core/common/src/main/java/alluxio/security/authentication/ChannelAuthenticator.java

```diff
@@ -106,39 +109,53 @@ public ChannelAuthenticator(String userName, String password, String impersonati
    */
   public Channel authenticate(ManagedChannel managedChannel, AlluxioConfiguration conf)
       throws AlluxioStatusException {
+    LOG.debug("Channel authentication initiated. ChannelId:{}, AuthType:{}, Target:{}", mChannelId,
+        mAuthType, managedChannel.authority());
+
     if (mAuthType == AuthType.NOSASL) {
       return managedChannel;
     }
 
-    // Create a channel for talking with target host's authentication service.
-    // Create SaslClient for authentication based on provided credentials.
-    SaslClient saslClient;
-    if (mUseSubject) {
-      saslClient =
-          SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mParentSubject, conf);
-    } else {
-      saslClient = SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mUserName,
-          mPassword, mImpersonationUser);
+    try {
+      // Create a channel for talking with target host's authentication service.
+      // Create SaslClient for authentication based on provided credentials.
+      SaslClient saslClient;
+      if (mUseSubject) {
+        saslClient = SaslParticipantProvider.Factory.create(mAuthType)
+            .createSaslClient(mParentSubject, conf);
+      } else {
+        saslClient = SaslParticipantProvider.Factory.create(mAuthType).createSaslClient(mUserName,
+            mPassword, mImpersonationUser);
+      }
+
+      // Create authentication scheme specific handshake handler.
+      SaslHandshakeClientHandler handshakeClient =
+          SaslHandshakeClientHandler.Factory.create(mAuthType, saslClient);
+      // Create driver for driving sasl traffic from client side.
+      SaslStreamClientDriver clientDriver =
+          new SaslStreamClientDriver(handshakeClient, mGrpcAuthTimeoutMs);
+      // Start authentication call with the service and update the client driver.
+      StreamObserver<SaslMessage> requestObserver =
+          SaslAuthenticationServiceGrpc.newStub(managedChannel).authenticate(clientDriver);
+      clientDriver.setServerObserver(requestObserver);
+      // Start authentication traffic with the target.
+      clientDriver.start(mChannelId.toString());
+      // Authentication succeeded!
+      // Attach scheme specific interceptors to the channel.
+
+      Channel authenticatedChannel =
+          ClientInterceptors.intercept(managedChannel, getInterceptors(saslClient));
+      return authenticatedChannel;
+    } catch (Exception exc) {
+      String message = String.format(
+          "Channel authentication failed. ChannelId: %s, AuthType: %s, Target: %s, Error: %s",
+          mChannelId, mAuthType, managedChannel.authority(), exc.getMessage());
```

Comment:
nit: prefer `exc.toString()`. It's like `exc.getMessage()` but includes the exception type. Some types of exceptions have empty messages because their type is enough to say what they are, e.g. `NullpointerException`